src/main/TopoMod.F90

```diff
@@ -240,18 +249,52 @@ subroutine UpdateTopo(this, bounds, num_icec, filter_icec, &
          this%topo_col(begc:endc), &
          this%needs_downscaling_col(begc:endc))
 
+    ! calculate area-weighted mean hillslope elevation on each landunit
+    if (use_hillslope) then
+       allocate(mean_hillslope_elevation(bounds%begl:bounds%endl))
+       mean_hillslope_elevation(:) = 0._r8
+       do l = bounds%begl, bounds%endl
+          mhe_norm = 0._r8
+          do c = lun%coli(l), lun%colf(l)
+             if (col%is_hillslope_column(c)) then
+                mean_hillslope_elevation(l) = mean_hillslope_elevation(l) &
+                     + col%hill_elev(c)*col%hill_area(c)
+                mhe_norm = mhe_norm + col%hill_area(c)
+             endif
+          enddo
+          if (mhe_norm > 0) then
+             mean_hillslope_elevation(l) = mean_hillslope_elevation(l)/mhe_norm
+          endif
+       enddo
+    endif
+       
     ! For any point that isn't downscaled, set its topo value to the atmosphere's
     ! topographic height. This shouldn't matter, but is useful if topo_col is written to
     ! the history file.
     !
     ! This could operate over a filter like 'allc' in order to just operate over active
     ! points, but I'm not sure that would speed things up much, and would require passing
     ! in this additional filter.
+
     do c = bounds%begc, bounds%endc
        if (.not. this%needs_downscaling_col(c)) then
+          ! For any point that isn't already set to be downscaled, set its topo value to the
```

Comment:
- [ ] Thanks a lot for this fix to TopoMod. Can you please also remove the now-redundant and incorrect comment a few lines up (the "For any point" comment above the loop... you can keep the "This could operate" paragraph, though).